sdk/core/core-client-rest/src/getClient.ts

```diff
@@ -153,18 +155,44 @@ export function getClient(
   };
 }
 
-function buildSendRequest(
+function buildOperation(
   method: HttpMethods,
-  baseUrl: string,
-  path: string,
+  url: string,
   pipeline: Pipeline,
-  requestOptions: RequestParameters = {},
-  args: string[] = [],
+  options: RequestParameters,
+  allowInsecureConnection?: boolean,
   httpClient?: HttpClient
-): Promise<HttpResponse> {
-  // If the client has an api-version and the request doesn't specify one, inject the one in the client options
-  const url = buildRequestUrl(baseUrl, path, args, requestOptions);
-  return sendRequest(method, url, pipeline, requestOptions, httpClient);
+): StreamableMethod {
+  return {
+    then: async function (onFulfilled) {
+      const result = await sendRequest(
```

Comment:
Does this propagate errors correctly? If not, we could fall back to being boring:

```ts
then: function (onFullfilled, onError) {
   try { 
      sendRequest().then(onFulfilled, onError);
   }
   catch (e) {
      onError(e);
   }
}

```

Reply:
Added a couple of tests throwing an error in the mocked `https` request and `fetch` and it is being propagated correctly